### ISSUE TITLE
[Error] JSX functional component

### ISSUE THREAD

**Issue by oreqizer** [eslint/typescript-eslint-parser#103]
**What version of TypeScript are you using?**
^2.0.3

**What version of `typescript-eslint-parser` are you using?**
^0.4.0

**What code were you trying to parse?**

<img width="289" alt="screen shot 2016-10-09 at 19 51 00" src="https://cloud.githubusercontent.com/assets/11172408/19222484/f1296294-8e59-11e6-8fed-53fd6025023f.png">

**What did you expect to happen?**
I expected it to lint the file.

**What happened?**
[npm-debug.txt](https://github.com/eslint/typescript-eslint-parser/files/518468/npm-debug.txt)

**Comment by JamesHenry** [comment]
Note: We do not yet officially support TypeScript 2 (although that does not mean that it does not work in all cases).

It is not clear from your debug output that this parser, or even ESLint, is the cause of your issue. Please provide more information as to what is going wrong.

What is your ESLint config?

**Comment by oreqizer** [comment]
```json
{
  "parser": "typescript-eslint-parser",
  "extends": ["airbnb"],
  "env": {
    "node": true,
    "jasmine": true
  },
  "globals": {
    "jest": true
  },
  "rules": {
    // conflicts with 'arrow body style'
    "no-confusing-arrow": "off",
    // due to TypeScript
    "no-undef": "off",
    "import/no-unresolved": "off",
    "import/extensions": "off",
    "react/prop-types": "off"
  }
}
```

**Comment by JamesHenry** [comment]
Please provide actual source code as per https://github.com/eslint/typescript-eslint-parser/issues/102#issuecomment-252653898

**Comment by oreqizer** [comment]
😅 sorry for me being non-predictive. I didn't take you wanting to copy the code into consideration

```javascript
const Component = props => (
  <div>
    <input
        type="text"
        value={props.input.value}
        onChange={props.input.onChange}
        onFocus={props.input.onFocus}
        onBlur={props.input.onBlur}
    />
  </div>
);
```

**Comment by JamesHenry** [comment]
Thanks @oreqizer, as per https://github.com/eslint/typescript-eslint-parser/issues/105 TS 2 seems to have broken some JSX related assertions, so my top priority is getting to the bottom of those so we can fully support TS 2.

Hopefully this issue you are having will be solved by that, but if not, I will loop back to this as soon as I am done with TS 2 support.

**Comment by soda0289** [comment]
@JamesHenry This should also be fixed by #158

**Comment by JamesHenry** [comment]
We have just released `2.0.0` to support the latest stable TypeScript release.

@oreqizer Please could you see if you still have the issue with that version?

**Comment by JamesHenry** [comment]
Closing as no feedback to the contrary was given. Please feel free to reopen if it is still an issue.
